contracts/utils/VersionOracle.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier:MIT
+pragma solidity ^0.6.2;
+pragma experimental ABIEncoderV2;
+// solhint-disable not-rely-on-time
+
+import "../interfaces/IVersionOracle.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract VersionOracle is IVersionOracle, Ownable {
+
+    mapping(bytes32 => bytes32) public idToVersion;
+    mapping(bytes32 => mapping(bytes32 => bytes32)) public versionHistory;
+    mapping(bytes32 => mapping(bytes32 => VersionInfo)) public values;
```

Comment:
values what? how is it a two-dimensional mapping?

Reply:
for each ID we need to know the latest version (that's idToVersion)
we need a linked list of all versions of a given ID - that is versionHistory
values are the actual data (and metadata) for each version

Reply:
removed on-chain logic (even though it was harder to implement and test in JS than it was in solidity...)